### PR TITLE
feat: add pretty debug information for common items

### LINKED ISSUE #85: Debug and/or Display traits for common items such as TraceWitness and Assignment
This is good for debugging. Add unit tests when appropriate.

### PR DESCRIPTION
**Pull request by dyxushuai**
Resolve #85

- [x] TraceWitness 
- [x] Assignments

### CODE REVIEW

**Comment by dyxushuai** [comment]
@leolara @ChihChengLiang I am keen to know if we only need print information with well indentation. Just like what format's `{:#}` flag's do 
https://doc.rust-lang.org/std/fmt/#sign0

**Comment by leolara** [comment]
I think we should make display look more like text, so for example

```
00(9): a = 1, b =2
01(10): a = 1, b =2
```

offset(step_uuid): assignations

where offset and step_uuid are hexadecimal.

**Comment by dyxushuai** [comment]
> I think we should make display look more like text, so for example
> 
> ```
> 00(9): a = 1, b =2
> 01(10): a = 1, b =2
> ```
> 
> offset(step_uuid): assignations
> 
> where offset and step_uuid are hexadecimal.

Ok, Make sense

File: src/wit_gen.rs [hunk 1]
```diff
@@ -39,6 +39,44 @@ pub struct TraceWitness<F> {
...
+                |mut acc, (queriable, value)| {
+                    acc.push_str(&format!("{:?} = {:?}, ", queriable, value));
+                    acc
+                },
+            );
+            // get the decimal width based on the assignments size, and extra one leading zero
+            // len in 0..=9 -> width = 2
+            // len in 10..=99-> width = 3
+            // len in 100..=999 -> width = 4
+            let decimal_width = step_instance
```
Comment (dyxushuai) [inline src/wit_gen.rs:60]: The output of the length of offset's padding is dynamic, does this make sense?

Comment (dyxushuai) [reply src/wit_gen.rs:60]: Or, other fields like `step_uuid` needs  to be dynamic width.

**Comment by dyxushuai** [comment]
After the format of `TraceWitness` is OK, I will continue work

**Comment by qwang98** [comment]
Just bumping this up. @leolara

**Comment by leolara** [comment]
@dyxushuai continue, I think it is ok, perhaps we don't need so detailed commenting, just showing the expected output.

**Comment by leolara** [comment]
@dyxushuai can you continue with this or is it finished?

**Comment by dyxushuai** [comment]
@leolara Pls take another look :)

**Review by leolara — APPROVED** [review]

**Pull request merged**